src/Workspaces/Core/Portable/Diagnostics/DiagnosticData.cs

```diff
@@ -263,13 +263,24 @@ private static ImmutableArray<DiagnosticDataLocation> GetAdditionalLocations(Tex
         {
             if (location.IsInSource)
             {
-                builder.AddIfNotNull(CreateLocation(document.Project.Solution.GetDocument(location.SourceTree), location));
+                builder.Add(CreateLocation(document.Project.Solution.GetDocument(location.SourceTree), location));
```

Comment:
`CreateLocation` always returns non-null. So switched AddIfNotNull to simply Add.